src/sidebar/util/oauth-client.js

```diff
@@ -182,19 +180,15 @@ export default class OAuthClient {
     });
 
     // Authorize user and retrieve grant token
-    let authUrl = this.authorizationEndpoint;
-    authUrl +=
-      '?' +
-      queryString.stringify({
-        client_id: this.clientId,
-        origin: $window.location.origin,
-        response_mode: 'web_message',
-        response_type: 'code',
-        state,
-      });
+    const authURL = new URL(this.authorizationEndpoint);
+    authURL.searchParams.set('client_id', this.clientId);
+    authURL.searchParams.set('origin', $window.location.origin);
+    authURL.searchParams.set('response_mode', 'web_message');
+    authURL.searchParams.set('response_type', 'code');
+    authURL.searchParams.set('state', state);
 
     // @ts-ignore - TS doesn't know about `location = <string>`.
-    authWindow.location = authUrl;
+    authWindow.location = authURL.toString();
```

Comment:
The `Window.location` setter is weird in that the getter and setter have different types with nothing in common (`Location` and `string` respectively). TypeScript doesn't support this. In a same-origin window one can use `location.href = <string>` instead, but this doesn't work for the cross-origin use case.

Reply:
That's...interesting. I wonder if it's worth adding to a commit message or otherwise persisting this information in an easier-to-find spot than this PR comment.